pygmt/datasets/samples.py

```diff
@@ -290,3 +217,141 @@ def _load_earth_relief_holes():
     """
     fname = which("@earth_relief_20m_holes.grd", download="c")
     return load_dataarray(fname, engine="netcdf4")
+
+
+class GMTSampleData(NamedTuple):
+    """
+    Information of a sample dataset.
+
+    Attributes
+    ----------
+    func : callable
+        The function that loads the sample dataset.
+    description : str
+        The description of the sample dataset.
+    """
+
+    func: Callable
+    description: str
+
+
+datasets = {
+    "bathymetry": GMTSampleData(
+        func=_load_baja_california_bathymetry,
+        description="Table of ship bathymetric observations off Baja California",
+    ),
+    "earth_relief_holes": GMTSampleData(
+        func=_load_earth_relief_holes,
+        description="Regional 20 arc-minutes Earth relief grid with holes",
+    ),
+    "fractures": GMTSampleData(
+        func=_load_fractures_compilation,
+        description="Table of hypothetical fracture lengths and azimuths",
+    ),
+    "hotspots": GMTSampleData(
+        func=_load_hotspots,
+        description="Table of locations, names, and symbol sizes of hotpots from "
+        "Mueller et al., 1993",
```

Comment:
To have a consistent citation style.
```suggestion
    "hotspots": GMTSampleData(
        func=_load_hotspots,
        description="Table of locations, names, and symbol sizes of hotpots from "
        "Mueller et al. (1993)",
```